File: Python/8py 2/move.py
"""
|--------------------------------------------------------------------------
| Grasshopper - Terminal game move function
|--------------------------------------------------------------------------
|
"""

move = lambda _, __: __+_+__


def move(position, roll):
    return position + roll * 2


def move(position, roll):
    return position+(roll<<1)


def move(a, b):
    return move(a + 2, b - 1) if b else a


print(move(0, 4), 8)
print(move(3, 6), 15)
print(move(2, 5), 12)

File: JavaScript/8js 3/move.js
/*
|--------------------------------------------------------------------------
| Grasshopper - Terminal game move function
|--------------------------------------------------------------------------
|
*/

function move(position, roll) {
   return position + roll * 2;
}


const move = (_, $, _$) => (_ + [_$ = -~[], ++_$][-~[]] * $)


function move(position, roll) {
   return position + roll + roll;
}


const move = (_, __) => _ + 2 * (__)


console.log(move(0, 4), 8);
console.log(move(3, 6), 15);
console.log(move(2, 5), 12);